input/postgres/relation_stats_ext.go

```diff
@@ -51,3 +51,80 @@ func handleRelationStatsExt(db *sql.DB, relStats state.PostgresRelationStatsMap,
 
 	return relStats, nil
 }
+
+const citusIndexSizeSQL = `
+WITH dist_idx_shard_stats_raw AS (
+	SELECT
+		dp.logicalrelid::regclass,
+		(pg_catalog.run_command_on_shards(logicalrelid::regclass::text, $$
+			SELECT
+				jsonb_agg(shard_idx_stats)
+			FROM (
+				SELECT
+					relnamespace::regnamespace AS idx_shard_schema,
+					indexrelid::regclass AS idx_shard_name,
+					COALESCE(pg_catalog.pg_relation_size(indexrelid), 0) AS idx_shard_bytes
+				FROM
+					pg_stat_user_indexes pgsui INNER JOIN pg_class pgc ON pgc.oid = pgsui.relid
+				WHERE
+					relid = '%s'::regclass
+			) AS shard_idx_stats
+		$$)).*
+	FROM
+		pg_dist_partition dp INNER JOIN pg_catalog.pg_class c ON (dp.logicalrelid::oid = c.oid)
+			 INNER JOIN pg_catalog.pg_namespace n ON (c.relnamespace = n.oid)
+	WHERE
+		($1 = '' OR (n.nspname || '.' || c.relname) !~* $1)
+), dist_idx_shard_stats AS (
+	SELECT
+		shardid,
+		bool_and(success) OVER() AS all_success,
+		jsonb_array_elements(result::jsonb) AS shard_info
+	FROM
+		dist_idx_shard_stats_raw
+)
+SELECT
+	pgc.oid,
+	sum((shard_info ->> 'idx_shard_bytes')::bigint) AS total_size_bytes
+FROM
+  pg_class pgc INNER JOIN dist_idx_shard_stats ON (
+		pgc.relkind = 'i'
+			AND pgc.relnamespace::regnamespace::text = (shard_info ->> 'idx_shard_schema')
+			AND pgc.relname = pg_catalog.regexp_replace(shard_info ->> 'idx_shard_name', '_' || shardid || '$', '')
```

Comment:
As far as I can tell, there's not a more direct way to match a shard index to its parent distributed index, so I'm relying on this approach.

Reply:
Yeah, I think that seems like the right way to go.